x/axelarnet/keeper/msg_server.go

```diff
@@ -285,9 +264,21 @@ func (s msgServer) RegisterAsset(c context.Context, req *types.RegisterAssetRequ
 		return &types.RegisterAssetResponse{}, fmt.Errorf("chain '%s' not found", req.Chain)
 	}
 
-	s.nexus.RegisterAsset(ctx, chain, req.Asset.Denom)
-	s.nexus.RegisterAsset(ctx, exported.Axelarnet, req.Asset.Denom)
-	s.BaseKeeper.RegisterAssetToCosmosChain(ctx, req.Asset, req.Chain)
+	if _, found := s.BaseKeeper.GetCosmosChainByName(ctx, req.Chain); !found {
+		return &types.RegisterAssetResponse{}, fmt.Errorf("chain '%s' is not a cosmos chain", req.Chain)
+	}
+
+	// register asset in chain state
+	err := s.nexus.RegisterAsset(ctx, chain, req.Asset)
+	if err != nil {
+		return nil, err
+	}
+
+	// also register on axelarnet, it routes assets from cosmos chains to evm chains
+	err = s.nexus.RegisterAsset(ctx, exported.Axelarnet, req.Asset)
```

Comment:
If `req.Asset` is native to chain, it is going to be native to axelarnet; If `req.Asset` is non-native to chain, it is going to be non-native to axelarnet. Is this expected?

Reply:
nope, good catch